lib/license_finder/package_utils/pypi.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'net/http'
+
+module LicenseFinder
+  class PyPI
+    CONNECTION_ERRORS = [
+      EOFError,
+      Errno::ECONNREFUSED,
+      Errno::ECONNRESET,
+      Errno::ECONNRESET,
+      Errno::EHOSTUNREACH,
+      Errno::EINVAL,
+      Net::OpenTimeout,
+      Net::ProtocolError,
+      Net::ReadTimeout,
+      OpenSSL::OpenSSLError,
+      OpenSSL::SSL::SSLError,
+      SocketError,
+      Timeout::Error
+    ].freeze
+
+    class << self
+      def definition(name, version)
+        response = request("https://pypi.org/pypi/#{name}/#{version}/json")
```

Comment:
I'm tempted to extract `pypi.org` as a source parameter so that this option can be customized to point to a `pypi.org` mirror.
Something like:

```ruby
def definition(name, version, source: 'pypi.org')
```
as a possible starting point. Then eventually connecting a CLI option to utilize this.

Reply:
Is this something that people actually do? I don't mind either way but for a first pass at least this is fine.

Reply:
Yep. In some organizations packages can only be pulled from an internal mirror where package analysis can be performed on the mirror. This is also true in some air gap scenarios.

Python even has a guide on setting up a mirror. https://packaging.python.org/guides/index-mirrors-and-caches/#id5

I'm going to skip this change because it's a little more nuanced than specifying a single source. The `Pipfile.lock` meta section can actually specify multiple sources to try to pull a package from.

As a starting point, I think defaulting to `pypi.org` is probably a good starting point. Also, this PR is getting rather large.